src/main/java/org/prebid/server/bidder/openweb/OpenWebBidder.java

```diff
@@ -0,0 +1,181 @@
+import org.prebid.server.json.JacksonMapper;
+import org.prebid.server.proto.openrtb.ext.ExtPrebid;
+import org.prebid.server.proto.openrtb.ext.request.openweb.ExtImpOpenweb;
+import org.prebid.server.proto.openrtb.ext.response.BidType;
+import org.prebid.server.util.BidderUtil;
+import org.prebid.server.util.HttpUtil;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class OpenWebBidder implements Bidder<BidRequest> {
+
+    private static final TypeReference<ExtPrebid<?, ExtImpOpenweb>> OPENWEB_EXT_TYPE_REFERENCE =
+            new TypeReference<ExtPrebid<?, ExtImpOpenweb>>() {
+            };
+
+    private final JacksonMapper mapper;
+    private final String endpointUrl;
+
+    public OpenWebBidder(String endpointUrl, JacksonMapper mapper) {
+        this.endpointUrl = Objects.requireNonNull(HttpUtil.validateUrl(endpointUrl));
+        this.mapper = Objects.requireNonNull(mapper);
+    }
+
+    @Override
+    public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest request) {
+        final Map<Integer, List<Imp>> sourceIdToModifiedImp = new HashMap<>();
+        final List<BidderError> errors = new ArrayList<>();
+
+        for (Imp imp : request.getImp()) {
+            try {
+                final ExtImpOpenweb extImpOpenweb = parseImpExt(imp);
+                final Integer sourceId = extImpOpenweb.getSourceId();
+                final Imp modifiedImp = modifyImp(imp, extImpOpenweb);
+
+                if (sourceIdToModifiedImp.containsKey(sourceId)) {
+                    sourceIdToModifiedImp.get(sourceId).add(modifiedImp);
+                } else {
+                    sourceIdToModifiedImp.put(sourceId, new ArrayList<>(Collections.singletonList(modifiedImp)));
+                }
+            } catch (PreBidException e) {
+                errors.add(BidderError.badInput(e.getMessage()));
+            }
+        }
+
+        if (sourceIdToModifiedImp.isEmpty()) {
+            return Result.withErrors(errors);
+        }
+        return Result.of(makeGroupRequests(request, sourceIdToModifiedImp), errors);
+    }
+
+    private ExtImpOpenweb parseImpExt(Imp imp) {
+        try {
+            return mapper.mapper().convertValue(imp.getExt(), OPENWEB_EXT_TYPE_REFERENCE).getBidder();
+        } catch (IllegalArgumentException e) {
+            throw new PreBidException(
+                    String.format(
+                            "ignoring imp id=%s, error while encoding impExt, err: %s",
+                            imp.getId(),
+                            e.getMessage()));
+        }
+    }
+
+    private Imp modifyImp(Imp imp, ExtImpOpenweb impExt) {
+        final ObjectNode modifiedImpExt = mapper.mapper().createObjectNode()
+                .set("openweb", mapper.mapper().valueToTree(impExt));
```

Comment:
Not sure we need to re-package `imp.ext`.
Probably `.set("openweb", imp.getExt)` is enough.

Reply:
It's because before modifyImp imp.ext contains bidder and prebid fields, and after marshalling we will have imp.ext.openweb instead imp.ext.openweb.[bidder, prebid]